flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -230,10 +253,12 @@ struct CharacterOpsBuilderImpl {
       assert(shape.size() == 1 && "only scalar character supported");
       // Materialize length for usage into character manipulations.
       auto len = builder.createIntegerConstant(lenType, shape[0]);
+      // FIXME: this seems to work for tests, but don't think it is correct
+      if (auto load = dyn_cast<fir::LoadOp>(character.getDefiningOp()))
```

Comment:
If I remember correctly, for CHARACTER assignment the bridge evaluates both the left and right side of the assignment to get their values. (Normally, the left-hand side is evaluated for an address.) So eventually it gets here and has a right-hand side value instead of the address. The hack to check for a LoadOp undoes the value and gets back the location to store to.

I think this all stems from trying to keep character variables in boxchars. More cleanup work there is needed.